8361536: [s390x] Saving return_pc at wrong offset

offamitkumar opened the pull request:
Fixes the bug where return pc was stored at a wrong offset, which causes issue with java abi. 

Issue appeared in #26004, see the comment: https://github.com/openjdk/jdk/pull/26004#issuecomment-3017928879.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8361536](https://bugs.openjdk.org/browse/JDK-8361536): [s390x] Saving return_pc at wrong offset (**Enhancement** - P3)


### Reviewers
 * [Lutz Schmidt](https://openjdk.org/census#lucy) (@RealLucy - **Reviewer**)
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/26209/head:pull/26209` \
`$ git checkout pull/26209`

Update a local copy of the PR: \
`$ git checkout pull/26209` \
`$ git pull https://git.openjdk.org/jdk.git pull/26209/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 26209`

View PR using the GUI difftool: \
`$ git pr show -t 26209`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/26209.diff">https://git.openjdk.org/jdk/pull/26209.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/26209#issuecomment-3055196247)
</details>

RealLucy reviewed (approved):
LGTM.
Thanks for fixing.

TheRealMDoerr reviewed (approved):
We're not really saving space. We just use less of the caller allocated stack space which is still as large as before.
But the change looks good and should make the stack walking code happy, because it can find the return_pc where it is expected, now.

offamitkumar commented:
@TheRealMDoerr I need reapproval. Can you provide one ? Thanks.

TheRealMDoerr reviewed (approved):
I did not request reverting it. I only corrected the wrong description. You can use your favorite offsets :-)

offamitkumar commented:
> I did not request reverting it. I only corrected the wrong description. You can use your favorite offsets :-)

Hi Martin, 
I reverted, because release build was crashing (I added crash log above). Mathematically, offset 80 should be fine to use. So I had to debug that.
I found out that mistakenly `Z_F14` was saved at offset `127` instead of `128`, which corrupted `Z_F13` register. That is fixed now. I am running test again.

offamitkumar commented:
> I am running test again.

I don't see new failure appearing in tier1 test with neither fastdebug nor release build. GHA failure are due to infra issue.
Will it be possible to get another approval as bots are not letting me integrate this :(

RealLucy reviewed (approved):
Good to go.

offamitkumar commented:
GHA failures are infra issue. Thanks for approval and reviews Lutz, Martin.

/integrate
